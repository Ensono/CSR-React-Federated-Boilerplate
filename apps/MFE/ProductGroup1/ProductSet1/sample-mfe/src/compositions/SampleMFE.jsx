import React from "runtime-library/lib/react"
import Button from "runtime-library/Button"
import Logo from "../components/Logo"
import Dialog from "../components/Dialog"

export class SampleMFE extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      dialogVisible: false,
    }
    this.handleClick = this.handleClick.bind(this)
    this.HanldeSwitchVisible = this.HanldeSwitchVisible.bind(this)
  }

  handleClick(ev) {
    console.log(ev)
    this.setState({
      dialogVisible: true,
    })
  }

  HanldeSwitchVisible(visible) {
    this.setState({
      dialogVisible: visible,
    })
  }

  render() {
    return (
      <div>
        <Logo />
        <br />
        <Button />
        <br />

        <button type="button" onClick={this.handleClick}>
          click to open dialog
        </button>
        <Dialog switchVisible={this.HanldeSwitchVisible} visible={this.state.dialogVisible} />
      </div>
    )
  }
}

export default SampleMFE
