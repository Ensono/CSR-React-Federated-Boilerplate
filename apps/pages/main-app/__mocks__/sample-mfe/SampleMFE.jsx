import React from 'react'
const SampleMFE = ()=><div>SampleMFE</div>
export default SampleMFE