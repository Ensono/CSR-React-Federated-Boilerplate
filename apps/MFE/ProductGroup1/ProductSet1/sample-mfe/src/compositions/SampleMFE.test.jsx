import React from "react"
import {render} from "@testing-library/react"
import SampleMFE from './SampleMFE'

describe("SampleMFE: ", () => {
    let props = {}

    it("should match the snapshot for type = primary", () => {
        const {asFragment} = render(
                <SampleMFE {...props} />
        )
        expect(asFragment()).toMatchSnapshot()
    })

    
})
