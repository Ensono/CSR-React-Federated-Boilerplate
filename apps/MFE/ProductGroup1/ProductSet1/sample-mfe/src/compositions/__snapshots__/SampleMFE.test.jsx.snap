// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SampleMFE:  should match the snapshot for type = primary 1`] = `
<DocumentFragment>
  <div>
    <img
      src="this is a file path"
      style="width: 500px; border-radius: 10px;"
    />
    <br />
    <button
      style="margin-left: 10px; color: rgb(255, 255, 255); background-color: rgb(64, 158, 255); padding: 12px 20px; font-size: 14px; border-radius: 4px; outline: none; border: 1px solid #dcdfe6; cursor: pointer;"
    >
      primary Button
    </button>
    <br />
    <button>
      click to open dialog
    </button>
  </div>
</DocumentFragment>
`;
